tests/phpunit/Differ/DiffColorizerTest.php

```diff
@@ -56,107 +57,136 @@ public function test_id_adds_colours_to_a_given_diff(string $originalDiff, strin
-                    CODE,
-                <<<'CODE'
-                    <code>
-                         protected function name()
-                         {
-                    <diff-del>-        return strtolower(get_class($this));</diff-del>
-                    <diff-add>+        strtolower(get_class($this));</diff-add>
-                    <diff-add>+        return null;</diff-add>
-                         }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a) {
+                <diff-del>-<diff-del-inline>        exit();</diff-del-inline></diff-del>
+                <diff-add>+</diff-add>
+                     }</code>
+                CODE,
+        ];
+
+        yield 'full-addition' => [
+            <<<'CODE'
+                     function ($a) {
+                -
+                +        exit();
+                     }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a) {
+                <diff-del>-</diff-del>
+                <diff-add>+<diff-add-inline>        exit();</diff-add-inline></diff-add>
+                     }</code>
+                CODE,
+        ];
+
+        yield 'partial-deletion' => [
+            <<<'CODE'
+                     function ($a) {
+                -        return 'foo' . 'bar';
+                +        return 'foo';
+                     }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a) {
+                <diff-del>-        return 'foo'<diff-del-inline> . 'bar'</diff-del-inline>;</diff-del>
+                <diff-add>+        return 'foo';</diff-add>
                      }</code>
-                    CODE,
-            ],
+                CODE,
+        ];
+
+        yield 'partial-addition' => [
+            <<<'CODE'
+                     function ($a) {
+                -        return 'foo';
+                +        return 'foo' . 'bar';
+                     }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a) {
+                <diff-del>-        return 'foo';</diff-del>
+                <diff-add>+        return 'foo'<diff-add-inline> . 'bar'</diff-add-inline>;</diff-add>
+                     }</code>
+                CODE,
+        ];
+
+        yield 'deletion-and-addition' => [
+            <<<'CODE'
+                     function ($a, $b) {
+                -        return $a && $b;
+                +        return $a || $b;
+                     }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a, $b) {
+                <diff-del>-        return $a <diff-del-inline>&&</diff-del-inline> $b;</diff-del>
+                <diff-add>+        return $a <diff-add-inline>||</diff-add-inline> $b;</diff-add>
+                     }</code>
+                CODE,
+        ];
+
+        yield 'bug-1999' => [
```

Comment:
Maybe having the link directly would be nice too?

```suggestion
        // https://github.com/infection/infection/issues/1999
        yield 'bug-1999' => [
```

Reply:
added